src/devices.coffee

```diff
@@ -30,13 +30,21 @@ module.exports =
         client.post 'device/', data, (err, res, body) ->
             if err
                 callback err
+            else if body.error is 'This name is already used'
+                deviceName = "#{options.deviceName}-2"
+                deviceName = deviceName.replace /-(\d+)-2/, (_, suffix) ->
```

Comment:
We make the same in mobile but it's more readable https://github.com/cozy/cozy-mobile/blob/master/www-src/app/replicator/main.coffee#L138, no ?